### DISCUSSION TITLE
cortex_query_frontend_non_step_aligned_queries_total > 0 when -query-frontend.align-queries-with-step=true

### DISCUSSION THREAD

**Discussion by agardiman** [grafana/mimir#9256] [Help and support] [answered]
Hi, 
the query `rate(cortex_query_frontend_non_step_aligned_queries_total[5m])` shows always positive numbers, however my expectation is for that number to be always zero since the option `-query-frontend.align-queries-with-step` is set to true for the query-frontend and ruler-query-frontend.

Is my expectation wrong? If so, how is that metric related to the cli option that I'm setting? If it's not wrong, what could be the issue?

Thanks

**Comment by 56quarters** [comment, marked as answer]
Your expectation is wrong though the metrics are confusingly named. `cortex_query_frontend_non_step_aligned_queries_total` is the number of incoming queries that aren't step aligned regardless of if they are adjusted to be step aligned afterwards. There's also `cortex_query_frontend_queries_step_aligned_total` which is the number of queries that are later adjusted to be step-aligned (if enabled globally or per-tenant).

**Reply by agardiman** [reply]
Ah that makes sense now! I can see that the 2 metrics are the same, so all good then.
Thank you very much Nick! :)
